jfxtras-agenda/src/main/java/jfxtras/internal/scene/control/skin/agenda/base24hour/AgendaSkinTimeScale24HourAbstract.java

```diff
@@ -238,7 +241,15 @@ public void refresh() {
 		refreshLocale();
 		setupAppointments();
 		nowUpdateRunnable.run(); 
-	}
+	}
+
+	/**
+	 * 
+	 */
+	@Override
+    public Pane getNodeForPopup(Appointment appointment) { return appointmentNodeMap.get(appointment); }
+    final private Map<Appointment, Pane> appointmentNodeMap = new WeakHashMap<>();
```

Comment:
I am only using the equals method for unit testing.  All considered, it is too much trouble.  I will replace it with the static isEqualsTo method as you suggested.  That will eliminate my need for it.

I am not sure which is the better option for the getNodeForPopup method.  I need to study System.identityHashCode(u) first.